cpp/src/arrow/compute/kernels/vector_nested.cc

```diff
@@ -76,6 +83,61 @@ struct ListParentIndicesArray {
 
   Status Visit(const LargeListType& type) { return VisitList(type); }
 
+  template <typename Type, typename offset_type = typename Type::offset_type>
+  Status VisitListView(const Type&) {
+    ArraySpan list_view{*input};
+
+    const offset_type* offsets = list_view.GetValues<offset_type>(1);
+    const offset_type* sizes = list_view.GetValues<offset_type>(2);
+    offset_type values_offset;
+    offset_type values_length;
+    ARROW_ASSIGN_OR_RAISE(std::tie(values_offset, values_length),
+                          RangeOfValuesUsed(list_view));
+
+    ARROW_ASSIGN_OR_RAISE(auto indices_validity,
+                          AllocateEmptyBitmap(values_length, ctx->memory_pool()));
+    auto* out_indices_validity = indices_validity->mutable_data();
+
+    ARROW_ASSIGN_OR_RAISE(auto indices, ctx->Allocate(values_length * sizeof(int64_t)));
+    auto* out_indices = indices->template mutable_data_as<int64_t>();
+    memset(out_indices, -1, values_length * sizeof(int64_t));
+
+    const auto* validity = list_view.GetValues<uint8_t>(0, 0);
+    RETURN_NOT_OK(arrow::internal::VisitSetBitRuns(
+        validity, list_view.offset, list_view.length,
+        [this, offsets, sizes, out_indices, out_indices_validity, values_offset](
+            int64_t run_start, int64_t run_length) {
+          for (int64_t i = run_start; i < run_start + run_length; ++i) {
+            const int64_t pop_count =
+                CountSetBits(out_indices_validity, offsets[i] - values_offset, sizes[i]);
+            if (ARROW_PREDICT_FALSE(pop_count > 0)) {
+              return Status::ExecutionError(
+                  "Function 'list_parent_indices' cannot produce parent indices for "
+                  "values used by more than one list-view array element.");
+            }
+            bit_util::SetBitmap(out_indices_validity, offsets[i] - values_offset,
+                                sizes[i]);
+            for (auto j = static_cast<int64_t>(offsets[i]);
+                 j < static_cast<int64_t>(offsets[i]) + sizes[i]; ++j) {
+              out_indices[j - values_offset] = i + base_output_offset;
+            }
+          }
+          return Status::OK();
+        }));
+
+    const int64_t null_count =
+        values_length - CountSetBits(out_indices_validity, 0, values_length);
```

Comment:
The null count could have easily been accumulated in the loop above, I'm not sure it makes sense to recompute it using a new popcount.

Reply:
Fixing.